File: src/Enterprise/VB.NET/patterns/LazyInitialization.vb
Imports System
Friend Module LazyInitializationExample
    Friend Function Run() As Boolean
        Dim builds=0,cache As String=Nothing
        Dim getValue As Func(Of String)=Function()
            If cache Is Nothing Then builds+=1 : cache="ready"
            Return cache
        End Function
        Return getValue()="ready" AndAlso getValue()="ready" AndAlso builds=1
    End Function
End Module

File: src/Enterprise/C#/patterns/LazyInitialization.cs
namespace Genkidama.PatternExamples;
public static class LazyInitializationExample { public static bool Run(){var builds=0;string? value=null;string Get()=>value??=Build();string Build(){builds++;return "ready";}return Get()=="ready"&&Get()=="ready"&&builds==1;} }
